### ISSUE TITLE
BUG: Leftover ClusterConfigurations and ClusterSummaries after CAPI cluster deletion

### ISSUE THREAD

**Issue by fchiacchiaretta** [projectsveltos/addon-controller#325]
# Problem Description

When deleting a CAPI cluster object from our management cluster, I can see that ClusterConfigurations and ClusterSummaries objects are not deleted.

# System Information

CLUSTERAPI VERSION: ` v1.5.1 `
SVELTOS VERSION: `v0.15.3`
KUBERNETES VERSION: `v1.27.5`

# Logs

I can see errors for failed reconciling of ClusterSummaries in addon-controller pod, while I couldn't find logs related to ClusterConfigurations objects.

```
I0915 10:07:50.539414       1 clustersummary_controller.go:140] "Reconciling" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="my-namespace/deploy-cilium-v1-26-capi-my-cluster" namespace="my-namespace" name="deploy-cilium-v1-26-capi-my-cluster" reconcileID="54854682-127f-4bb4-8806-23fcab276c57"
I0915 10:07:50.539878       1 clustersummary_controller.go:211] "Reconciling ClusterSummary delete" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="my-namespace/deploy-cilium-v1-26-capi-my-cluster" namespace="my-namespace" name="deploy-cilium-v1-26-capi-my-cluster" reconcileID="54854682-127f-4bb4-8806-23fcab276c57"
I0915 10:07:50.539940       1 clusterproxy.go:100] "Cluster does not exist" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="my-namespace/deploy-cilium-v1-26-capi-my-cluster" namespace="my-namespace" name="deploy-cilium-v1-26-capi-my-cluster" reconcileID="54854682-127f-4bb4-8806-23fcab276c57"
E0915 10:07:50.539998       1 clustersummary_controller.go:224] "failed to remove ResourceSummary." err="Cluster my-namespace/my-cluster does not exist: Cluster.cluster.x-k8s.io \"my-cluster\" not found" controller="clustersummary" controllerGroup="config.projectsveltos.io" controllerKind="ClusterSummary" ClusterSummary="my-namespace/deploy-cilium-v1-26-capi-my-cluster" namespace="my-namespace" name="deploy-cilium-v1-26-capi-my-cluster" reconcileID="54854682-127f-4bb4-8806-23fcab276c57"
```

Here is the `deploy-cilium-v1.26` ClusterProfile

```
apiVersion: config.projectsveltos.io/v1alpha1
kind: ClusterProfile
metadata:
  name: deploy-cilium-v1-26
spec:
  clusterSelector: sveltos=enabled,kubernetes=v1-26
  helmCharts:
  - chartName: cilium/cilium
    chartVersion: 1.12.12
    helmChartAction: Install
    releaseName: cilium
    releaseNamespace: kube-system
    repositoryName: cilium
    repositoryURL: https://helm.cilium.io/
    values: |
      k8sServiceHost: "{{ .Cluster.spec.controlPlaneEndpoint.host }}"
      k8sServicePort: "{{ .Cluster.spec.controlPlaneEndpoint.port }}"
      hubble:
        enabled: false
      nodePort:
        enabled: true
      kubeProxyReplacement: strict
      operator:
        replicas: 1
        updateStrategy:
          rollingUpdate:
            maxSurge: 0
            maxUnavailable: 1
  reloader: false
  stopMatchingBehavior: WithdrawPolicies
  syncMode: Continuous
```

I'm not managing cluster via `sveltosctl`, but just creating and deleting cluster object on the management cluster.

**Comment by gianlucam76** [comment]
Thank you @fchiacchiaretta. Looking into this now.

**Comment by gianlucam76** [comment]
Thank you for reporting this bug. Issue was with ClusterSummaryReconciler which incorrectly assumed it had to cleanup resources in the cluster. Clearly when corresponding cluster is deleted and gone, there is nothing to cleanup there.

Fix was easy but it was a path somehow I did not consider. So thank you again for reporting this bug. 

[Fix](https://github.com/projectsveltos/addon-controller/pull/333) has been merged. 

Till new release with fix is out, workaround (if possible) is to remove the labels from cluster before deleting it so allowing ClusterSummary delete path to be triggered before cluster is gone.
